packages/cli/openapi-ir-to-fern/src/buildEnvironments.ts

```diff
@@ -1,79 +1,122 @@
+import { RawSchemas } from "@fern-api/yaml-schema";
 import { OpenApiIrConverterContext } from "./OpenApiIrConverterContext";
 
 const PRODUCTION_ENVNIRONMENT_NAME = "Production";
 const DEFAULT_ENVIRONMENT_NAME = "Default";
 
+function isMultiServerSchema(
+    schema: RawSchemas.EnvironmentSchema
+): schema is RawSchemas.MultipleBaseUrlsEnvironmentSchema {
+    return typeof schema === "object" && Object.hasOwn(schema, "urls");
```

Comment:
nit: First time I've seen `Object.hasOwn` for this - I'm sure this works just fine, but I wonder if it's worth making this consistent with the other `isRaw` helpers like the following:

```typescript
export function isRawObjectDefinition(rawTypeDeclaration: TypeDeclarationSchema): rawTypeDeclaration is ObjectSchema {
    return (
        (rawTypeDeclaration as ObjectSchema).extends != null || (rawTypeDeclaration as ObjectSchema).properties != null
    );
}
```

I imagine we could do something like:

```typescript
function isRawMultipleBaseUrlsEnvironmentSchema(schema: RawSchemas.EnvironmentSchema): schema is RawSchemas.MultipleBaseUrlsEnvironmentSchema {
    return schema as MultipleBaseUrlsEnvironmentSchema).urls != null;
}
```

We could also export this function in a similar place as `isRawObjectDefinition` so that it can be easily referenced by other callers later on. This could be in its own file like [this](https://github.com/fern-api/fern/blob/6f002b525a2e1e512b41fe77dc4acf6fa78a2390/packages/cli/yaml/yaml-schema/src/utils/isRawProtobufSourceSchema.ts#L3).

Reply:
Nice, it already existed, so imported it! Thanks for the tip